tests/DocBlock/TypeExpressionTest.php

```diff
@@ -253,6 +253,10 @@ public static function provideGetTypesCases(): iterable
 
         yield ['array{a: int, b: int, c: int, d: int, e: int, f: int, g: int, h: int, i: int, j: int, k: int, l: int, with-dash: int}'];
 
+        yield ['string'.str_repeat('[]', 128)];
```

Comment:
👍 for simple test case

I found some related https://github.com/doctrine/dbal/pull/5138 but I do not fully understand it. Maybe can you explain using https://regex101.com/? Or is it some JIT pcre issue?

Reply:
here is a fix: https://github.com/6b7562617765726c6f73/PHP-CS-Fixer/pull/2

it should prevent any backtracking of commontly used recursive parts :)